src/h5web/dataset-visualizer/DimensionMapper.tsx

```diff
@@ -54,40 +54,43 @@ function DimensionMapper(props: Props): JSX.Element {
     );
   }
 
-  function renderDimensionButton(dimension: 'x' | 'y'): JSX.Element {
+  function renderAxisMapper(axis: 'x' | 'y'): JSX.Element {
+    const selectedDim = mapperState.indexOf('x');
     return (
-      <div className={styles.dimensionButtonWrapper}>
-        <span className={styles.dimensionName}>{dimension}</span>
-        <ButtonGroup
-          ariaLabel={`Dimension as ${dimension}`}
-          className={styles.buttonGroup}
-          buttonClassName={styles.btn}
-          buttons={Object.keys(rawDims).map(dimKey => {
```

Comment:
I've used the compound components pattern: https://kentcdodds.com/blog/compound-components-with-react-hooks

`ToggleGroup` provides the currently selected value of the group through a local context, which its child `ToggleGroup.Btn` components read in order to determine whether they are the selected button. The group also provides the `onChange` callback function through context so the buttons can call it when they are clicked.